nextjs-with-typescript build

All the steps I followed is https://github.com/mui-org/material-ui/tree/master/examples/nextjs-with-typescript

```
curl https://codeload.github.com/mui-org/material-ui/tar.gz/master | tar -xz --strip=2  material-ui-master/examples/nextjs-with-typescript
cd nextjs-with-typescript
```

```
npm install
npm run build
```

And the errors show below:

```
    ERROR in X:/fake_project/nextjs-with-typescript/pages/_document.tsx
    69:17 Type '{} | ReactNodeArray | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)>[] | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.
        67 |     ...initialProps,
        68 |     // Styles fragment is rendered after the app and page rendering finish.
      > 69 |     styles: [...initialProps.styles, sheets.getStyleElement()],
           |                 ^
        70 |   };
        71 | };
        72 |

    > Build error occurred
    Error: > Build failed because of webpack errors
```

Is there any thing I need to notice when `build`? `npm run dev` is working properly, tho.

Tested `nextjs` without `Typescript` works, so there is probably something wrong with `Typescript`.

This fix the issue with normal build! But It's failed on turning on awp.

Add in code (page/index.tsx)
```
export const config = { amp: true };
```

Result:
```
Amp Validation

/  error  The mandatory attribute '⚡' is missing in tag 'html'.                 https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml#required-markup
   error  The tag 'meta name=viewport' appears more than once in the document.  https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml#required-markup

> Build error occurred
Error: AMP Validation caused the export to fail. https://err.sh/zeit/next.js/amp-export-validation
    at _default (D:\Project\McDedicatedServer\minecraft_scepter_home\innerTest\nextjs-with-typescript\node_modules\next\dist\export\index.js:15:606)
    at async build (D:\Project\McDedicatedServer\minecraft_scepter_home\innerTest\nextjs-with-typescript\node_modules\next\dist\build\index.js:18:145)
Creating an optimized production build .Automatically optimizing pages .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nextjs-with-typescript@4.0.0 build: `next build`
npm ERR! Exit status 1
```

I'm not sure If It's safe to pull request. However, this change fix the Typescript error on normal build.

--- a/docs/pages/_document.js
+++ b/docs/pages/_document.js
@@ -144,7 +144,7 @@ MyDocument.getInitialProps = async ctx => {
     canonical: pathnameToLanguage(ctx.req.url).canonical,
     userLanguage: ctx.query.userLanguage || 'en',
     styles: [
-      ...initialProps.styles,
+      ...React.Children.toArray(initialProps.styles),
       <style
         id="jss-server-side"
         key="jss-server-side"

--- a/examples/nextjs-with-typescript/pages/_document.tsx
+++ b/examples/nextjs-with-typescript/pages/_document.tsx
@@ -66,6 +66,6 @@ MyDocument.getInitialProps = async ctx => {
   return {
     ...initialProps,
     // Styles fragment is rendered after the app and page rendering finish.
-    styles: [...initialProps.styles, sheets.getStyleElement()],
+    styles: [...React.Children.toArray(initialProps.styles), sheets.getStyleElement()],
   };
 };

--- a/examples/nextjs/pages/_document.js
+++ b/examples/nextjs/pages/_document.js
@@ -66,6 +66,6 @@ MyDocument.getInitialProps = async ctx => {
   return {
     ...initialProps,
     // Styles fragment is rendered after the app and page rendering finish.
-    styles: [...initialProps.styles, sheets.getStyleElement()],
+    styles: [...React.Children.toArray(initialProps.styles), sheets.getStyleElement()],
   };
 };

